internal/v2/fakes2av2/fakes2av2.go

```diff
@@ -98,15 +99,19 @@ func offloadPrivateKeyOperation(req *s2av2pb.OffloadPrivateKeyOperationReq, host
 			s := fmt.Sprintf("Invalid hostname tied to SessionReq: %s", hostname)
 			return nil, errors.New(s)
 		}
-
-		// TODO(rmehta19): Parse req.GetSignatureAlgorithm() to find Hash function that
-		// should be used here. Currently hardcoded to use SHA256.
-
-		// Sign the SHA256 hash with the private key.
-		var opts crypto.Hash = crypto.SHA256
-		signedBytes, err := root.PrivateKey.(crypto.Signer).Sign(rand.Reader, req.GetInBytes(), opts)
-		if err != nil {
-			return nil, err
+		var signedBytes []byte
+		if req.GetSignatureAlgorithm() == s2av2pb.SignatureAlgorithm_S2A_SSL_SIGN_RSA_PKCS1_SHA256 {
+			var opts crypto.Hash = crypto.SHA256
+			signedBytes, err = root.PrivateKey.(crypto.Signer).Sign(rand.Reader, req.GetInBytes(), opts)
+			if err != nil {
+				return nil, err
+			}
+		} else if req.GetSignatureAlgorithm() == s2av2pb.SignatureAlgorithm_S2A_SSL_SIGN_RSA_PSS_RSAE_SHA256 {
+			var opts = rsa.PSSOptions { -1, crypto.SHA256 }
```

Comment:
readability nit: same here, I think it should be `opts := &rsa.PSSOptions{SaltLength: -1, Hash: crypto.SHA256}`?